src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -358,4 +408,132 @@ private function modelMethodIsRelationship($model, $method)
 
         return $method;
     }
+
+    /**
+     * This function is responsible for setting up the morph fields structure.
+     * Developer can define the morph structure as follows:
+     *  'morphOptions => [
+     *       ['nameOnAMorphMap', 'label', [options]],
+     *       ['App\Models\Model'], // display the name of the model
+     *       ['App\Models\Model', 'label', ['data_source' => backpack_url('smt')]
+     *  ]
+     * OR
+     * ->addMorphOption('App\Models\Model', 'label', ['data_source' => backpack_url('smt')]).
+     *
+     * @param  string|array  $fieldName
+     * @param  string  $key
+     * @param  string|null  $label
+     * @param  array  $options
+     * @return void|array
+     */
+    public function addMorphOption($fieldName, string $key, $label = null, array $options = [])
```

Comment:
Would work just as well as `protected` but disallow editing, correct?

```suggestion
    protected function addMorphOption($fieldName, string $key, $label = null, array $options = [])
```

Reply:
This can't be protected or private otherwise `CrudField` class couldn't call it. (CrudField is an "outsider" here, from our conversation on discord the other day).
ping @tabacitu

Reply:
Damn it! Ok...